external/loaders/loaders/mappers/_hybrid.py

```diff
@@ -215,3 +216,76 @@ def open_fine_resolution_nudging_hybrid(
             include_temperature_nudging=include_temperature_nudging,
         )
     )
+
+
+def open_fine_resolution_dataset(
+    fine_url: str = "gs://vcm-ml-experiments/default/2021-04-27/2020-05-27-40-day-X-SHiELD-simulation/fine-res-budget.zarr",  # noqa: E501
+    state_url: Optional[str] = None,
+    include_temperature_nudging: bool = False,
+) -> xarray.Dataset:
+
+    fine = open_zarr_maybe_consolidated(fine_url)
```

Comment:
There is a lot of copy paste here with the hybrid mapper above. There seem like some clear opportunities for resuse e.g. by defining some shared helpers like `assign_apparent_sources`. 

Also mixes I/O and computations. I would pull the zarr opening to the calling routine and making a pure function `assign_fine_res_sources` for the rest:
```
fine = open_zarr_maybe_consolidated(fine_url)
ds = assign_fine_res_sources(fine, **kwargs)
return XarrayMapper(ds)
```

Then bonus points for cleanly documenting the required variable names. You can probably use the protocol I defined here to do that: https://github.com/ai2cm/fv3net/blob/b9d97ea72c296b6b0eb830d4115599a2500f2496/external/loaders/loaders/mappers/_hybrid.py#L53

e.g.

```
def assign_fine_res_sources(fine: FineResSources) -> xr.Dataset
```

Reply:
Okay, I have reduced the copy paste as it relates to computing the fine-res sources and standardizing that dataset. I'd prefer not to make bigger changes (e.g. modifying I/O) since I think consistency with the `open_fine_resolution_nudging_hybrid_dataset` function is important.